pkg/kv/kvserver/store_split.go

```diff
@@ -317,19 +317,6 @@ func (s *Store) SplitRange(
 		return errors.Errorf("left range is not splittable by right range: %+v, %+v", oldLeftDesc, rightDesc)
 	}
 
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	if exRng, ok := s.mu.uninitReplicas[rightDesc.RangeID]; rightReplOrNil != nil && ok {
```

Comment:
This did not assert that `rightRepl` must be in the map. However, it must, because the replica is asserted to be uninitialized up the stack in `prepareRightReplicaForSplit`.

So, swapping this for a stricter check in `s.markReplicaInitializedLockedReplLocked` should be legit?

Reply:
Yes, de facto `raftMu` is locked in all the places that add to / delete from this map. Except one test :)
The `raftMu` lock requirement is also not explicitly documented in the method name.

I added a comment to `Store.uninitReplicas` field, but maybe it should be documented even better.